### ISSUE TITLE
Fix S4143: FN for Add followed by assignment

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-csharp#1908]
### Description

S4143 does not detect when a dictionary item as added via method invocation and then modified via an assignment.

### Expected behavior
The below snippet is taken from the rule RSPEC
```
towns.Add(y, "Boston");
towns[y] = "Paris"; // Noncompliant
```

### Actual behavior

```
towns.Add(y, "Boston");
towns[y] = "Paris"; // FN
```

* SonarC# Version 7.5
